templates/types/streaming/fastapi/app/api/routers/chat.py

```diff
@@ -44,16 +42,36 @@ async def chat(
         )
         for m in data.messages
     ]
+    return last_message, messages
 
-    # query chat engine
-    response = await chat_engine.astream_chat(lastMessage.content, messages)
 
-    # stream response
+@r.post("")
+async def chat(
+    request: Request,
+    data: _ChatData,
+    chat_engine: BaseChatEngine = Depends(get_chat_engine),
+):
+    last_message, messages = await preprocess_request(data)
+
+    response = await chat_engine.astream_chat(last_message.content, messages)
+
     async def event_generator():
         async for token in response.async_response_gen():
-            # If client closes connection, stop sending events
             if await request.is_disconnected():
                 break
             yield token
 
     return StreamingResponse(event_generator(), media_type="text/plain")
+
+
+@r.post("/request")
```

Comment:
```suggestion
# non-streaming endpoint - delete if not needed
@r.post("/request")
```

Reply:
@thucpn i think in the FastAPI code, it's easy to just delete the function endpoint, so we can leave it in one file for simplicity